tests/jerry/es2015/proxy_is_extensible.js

```diff
@@ -26,21 +25,77 @@ try {
   Object.isFrozen(proxy)
   assert(false);
 } catch (e) {
-  assert(e instanceof TypeError);
+  assert(e === 42);
 }
 
 try {
   // 7.3.15
   Object.isSealed(proxy)
   assert(false);
 } catch (e) {
-  assert(e instanceof TypeError);
+  assert(e === 42);
 }
 
 try {
   // 7.2.5
   Object.isExtensible(proxy)
   assert(false);
+} catch (e) {
+  assert(e === 42);
+}
+
+// no trap
+var target = {};
+var handler = {};
+var proxy = new Proxy(target, handler);
+assert(Object.isExtensible(target) === true)
+assert(Object.isExtensible(proxy) === true);
+
+// undefined trap
+var target = {};
+var handler =  { isExtensible: null };
+var proxy = new Proxy(target, handler);
+assert(Object.isExtensible(target) === true)
+assert(Object.isExtensible(proxy) === true);
+
+// invalid trap
+var target = {};
+var handler =  { isExtensible: true };
+var proxy = new Proxy(target, handler);
+
+try {
+  Object.isExtensible(proxy);
+  assert(false);
+} catch (e) {
+  assert(e instanceof TypeError); 
+}
+
+// valid trap
+var target = { prop1: true };
+var handler = {
+  isExtensible(target) {
+    target.prop1 = false;
+    return Object.isExtensible(target);
+  }
+};
+
+var proxy = new Proxy(target, handler);
+assert(Object.isExtensible(proxy) === true);
+assert(target.prop1 === false);
+
+// trap result is invalid
```

Comment:
Because by default, the target object is extensible so invoking the Object.isExtensible method with that target should return true. However, in the handler, we just simply return with false.